keycloak/authentication_execution.go

```diff
@@ -0,0 +1,144 @@
+
+	err := keycloakClient.get(fmt.Sprintf("/realms/%s/authentication/flows/%s/executions", realmId, parentFlowAlias), &authenticationExecutions, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return authenticationExecutions, err
+}
+
+func (keycloakClient *KeycloakClient) NewAuthenticationExecution(execution *AuthenticationExecution) error {
+	_, location, err := keycloakClient.post(fmt.Sprintf("/realms/%s/authentication/flows/%s/executions/execution", execution.RealmId, execution.ParentFlowAlias), &authenticationExecutionCreate{Provider: execution.Authenticator})
+	if err != nil {
+		return err
+	}
+
+	execution.Id = getIdFromLocationHeader(location)
+
+	err = keycloakClient.UpdateAuthenticationExecution(execution)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (keycloakClient *KeycloakClient) GetAuthenticationExecution(realmId, parentFlowAlias, id string) (*AuthenticationExecution, error) {
+	var authenticationExecution AuthenticationExecution
+
+	err := keycloakClient.get(fmt.Sprintf("/realms/%s/authentication/executions/%s", realmId, id), &authenticationExecution, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	authenticationExecution.RealmId = realmId
+	authenticationExecution.ParentFlowAlias = parentFlowAlias
+
+	return &authenticationExecution, nil
+}
+
+func (keycloakClient *KeycloakClient) UpdateAuthenticationExecution(execution *AuthenticationExecution) error {
+	authenticationExecutionUpdateRequirement := &authenticationExecutionRequirementUpdate{
+		RealmId:         execution.RealmId,
+		ParentFlowAlias: execution.ParentFlowAlias,
+		Id:              execution.Id,
+		Requirement:     execution.Requirement,
+	}
+	return keycloakClient.UpdateAuthenticationExecutionRequirement(authenticationExecutionUpdateRequirement)
+}
+
+func (keycloakClient *KeycloakClient) UpdateAuthenticationExecutionRequirement(executionRequirementUpdate *authenticationExecutionRequirementUpdate) error {
+	return keycloakClient.put(fmt.Sprintf("/realms/%s/authentication/flows/%s/executions", executionRequirementUpdate.RealmId, executionRequirementUpdate.ParentFlowAlias), executionRequirementUpdate)
+}
+
+func (keycloakClient *KeycloakClient) DeleteAuthenticationExecution(realmId, id string) error {
+	err := keycloakClient.delete(fmt.Sprintf("/realms/%s/authentication/executions/%s", realmId, id), nil)
+	if err != nil {
+		// For whatever reason, this fails sometimes with a 500 during acceptance tests. try again
+		return keycloakClient.delete(fmt.Sprintf("/realms/%s/authentication/executions/%s", realmId, id), nil)
+	}
+
+	return nil
+}
+
+func (keycloakClient *KeycloakClient) RaiseAuthenticationExecutionPriority(realmId, id string) error {
+	_, _, err := keycloakClient.post(fmt.Sprintf("/realms/%s/authentication/executions/%s/raise-priority", realmId, id), nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (keycloakClient *KeycloakClient) LowerAuthenticationExecutionPriority(realmId, id string) error {
```

Comment:
are these two functions used?  if not, how should users raise and lower the priority for executions?

Reply:
There are not used, but they do work.
I left in here in the case some one worked out a better solution, and could use these methods....

Should I remove them?

Reply:
No, I was just curious how the priority was determined, but then I realized in your example that you used `depends_on` in order to ensure subflows with higher priorities were created before subflows with lower priorities.

This approach makes sense, although it would be impossible to update this aftwards.  But if that's good enough for you then I'm okay with this.  I'm hoping that this is good enough for most users until Keycloak X is released, in which case I'll probably have to rewrite a lot of this provider anyways.